posthog/hogql/database/schema/event_sessions.py

```diff
@@ -0,0 +1,126 @@
+from copy import deepcopy
+from typing import Any, Dict, List
+from posthog.hogql import ast
+from posthog.hogql.context import HogQLContext
+from posthog.hogql.database.models import FieldOrTable, IntegerDatabaseField, StringDatabaseField, VirtualTable
+from posthog.hogql.parser import parse_select
+from posthog.hogql.resolver_utils import get_long_table_name, lookup_field_by_name
+from posthog.hogql.visitor import CloningVisitor, TraversingVisitor
+
+
+class EventsSessionSubTable(VirtualTable):
+    fields: Dict[str, FieldOrTable] = {
+        "$session_id": StringDatabaseField(name="$session_id"),
+        "session_duration": IntegerDatabaseField(name="session_duration"),
+    }
+
+    def to_printed_clickhouse(self, context):
+        return "events"
+
+    def to_printed_hogql(self):
+        return "events"
+
+
+class GetFieldsTraverser(TraversingVisitor):
+    fields: List[ast.Field]
+
+    def __init__(self, expr: ast.Expr):
+        super().__init__()
+        self.fields = []
+        super().visit(expr)
+
+    def visit_field(self, node: ast.Field):
+        self.fields.append(node)
+
+
+class WhereClauseExtractor:
+    compare_operators: List[ast.Expr]
+
+    def __init__(self, where_expression: ast.Expr, from_table_name: str, select_query_type: ast.SelectQueryType):
+        self.table_name = from_table_name
+        self.select_query_type = select_query_type
+        self.compare_operators = self.run(deepcopy(where_expression))
+
+    def _is_field_on_table(self, field: ast.Field) -> bool:
+        if len(field.chain) == 0:
+            return False
+
+        # If the field contains at least two parts, the first might be a table.
+        if len(field.chain) > 1:
+            type = self.select_query_type.tables[str(field.chain[0])]
+
+            name = get_long_table_name(self.select_query_type, type)
+            if name == self.table_name:
+                field.chain.pop(0)
+                return True
+            else:
+                return False
```

Comment:
This just checks that the table exists, but not the field...?

Reply:
Yes, you make an excellent point 🤦